lib/galaxy/webapps/galaxy/api/folders.py

```diff
@@ -246,20 +246,51 @@ def delete( self, trans, id, **kwd ):
         :returns:   detailed folder information
         :rtype:     dictionary
 
-        :raises: ItemAccessibilityException, MalformedId, ObjectNotFound
         """
-        folder = self.folder_manager.get( trans, self.folder_manager.cut_and_decode( trans, id ), True )
+        folder = self.folder_manager.get( trans, self.folder_manager.cut_and_decode( trans, encoded_folder_id ), True )
         undelete = util.string_as_bool( kwd.get( 'undelete', False ) )
         folder = self.folder_manager.delete( trans, folder, undelete )
         folder_dict = self.folder_manager.get_folder_dict( trans, folder )
+        folder_dict['update_time'] = folder.update_time.strftime( "%Y-%m-%d %I:%M %p" )
         return folder_dict
 
     @expose_api
-    def update( self, trans, id, library_id, payload, **kwd ):
+    def update( self, trans, encoded_folder_id, **kwd ):
         """
-        PUT /api/folders/{encoded_folder_id}
+        * PATCH /api/folders/{encoded_folder_id}
+           Updates the folder defined by an ``encoded_folder_id`` with the data in the payload.
+
+       .. note:: Currently, only admin users can update library folders. Also the folder must not be `deleted`.
+
+        :param  id:      the encoded id of the folder
+        :type   id:      an encoded id string
+
+        :param  payload: (required) dictionary structure containing::
+            'name':         new folder's name, cannot be empty
+            'description':  new folder's description
+        :type   payload: dict
 
+        :returns:   detailed folder information
+        :rtype:     dict
+
+        :raises: RequestParameterMissingException
         """
+        decoded_folder_id = self.folder_manager.cut_and_decode( trans, encoded_folder_id )
+        folder = self.folder_manager.get( trans, decoded_folder_id )
+
+        payload = kwd.get( 'payload', None )
+        if payload:
+            name = payload.get( 'name', None )
+            if not name:
+                raise exceptions.RequestParameterMissingException( "Parameter 'name' of folder is required. You cannot remove it." )
+            if payload.get( 'description', None ) or payload.get( 'description', None ) == '':
```

Comment:
I'm not sure this is correct. If `payload.get( 'description', None )` is None, this will leave `description` undefined, and a `NameError` exception will be raised at line 290.

I think this check can be simply removed.